src/reducers/Reports/index.js

```diff
@@ -2,13 +2,16 @@ import * as types from '@constants/Types';
 
 const INITIAL_STATE = {
     fetchingReport: false,
-    reportsList: []
+    reportsList: [],
+    selected: null
 }
 
 export default (state = INITIAL_STATE, action) => {
     switch (action.type) {
         case types.TOGGLE_FETCHING_STATUS: return { ...state, fetchingReport: !state.fetchingReport }
         case types.SET_CURRENCIES_DATA: return { ...state, reportsList: state.reportsList.concat(action.payload) }
+        case types.SET_SELECTED_DATA: return { ...state, selected: action.payload }
+        case types.DELETE_SELECTED_DATA: return { ...state, reportsList: state.reportsList.filter((r) => r!==state.selected), selected: null }

```

Comment:
In order to make clear separation of concerns in your application, I do suggest making separate actions for filtering reportsList (as above) and separate one for clearing out selected element. Reducers with multiple state transformation are harder to debug and Redux gets little bit less modular.

My suggestion is to create:

Redux action:

const clearSelectedItem = () => { type: types.CLEAR_SELECTED_ITEM }

(no value passed, since it just assigns null to selected Redux store variable)

and corresponding reducer, like so:

...
case types.CLEAR_SELECTED_ITEM: return { ...state, selected: null }
...

PS. Props for filter, it's a common and good practice to use it like above.